### PR TITLE
MINOR: Improve Connect docs

### PR DESCRIPTION
**Pull request by mimaison**
- Fix indendation of code blocks
- Add links to all SMTs and Predicates

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/connect.html [hunk 1]
```diff
@@ -41,7 +41,7 @@ <h4><a id="connect_running" href="#connect_running">Running Kafka Connect</a></h
     <p>In standalone mode all work is performed in a single process. This configuration is simpler to setup and get started with and may be useful in situations where only one worker makes sense (e.g. collecting log files), but it does not benefit from some of the features of Kafka Connect such as fault tolerance. You can start a standalone process with the following command:</p>
 
     <pre class="brush: bash;">
-    &gt; bin/connect-standalone.sh config/connect-standalone.properties connector1.properties [connector2.properties ...]
+&gt; bin/connect-standalone.sh config/connect-standalone.properties connector1.properties [connector2.properties ...]
     </pre>
```
Comment (showuon) [inline docs/connect.html]: I found we put an additional line for each command. 
image


We should put the ending `</pre>` at the end of the command, ex:
```
<pre class="brush: bash;">
&gt; bin/connect-standalone.sh config/connect-standalone.properties connector1.properties [connector2.properties ...]</pre>
```

Same as other commands.

**Review by showuon — COMMENTED** [review]
@mimaison , thanks for fixing it. It will make the doc look better! Left a comment. Thanks.

File: docs/connect.html (hunk 1 above)
Comment (mimaison) [reply docs/connect.html]: Good point! I've pushed an update

**Review by showuon — COMMENTED** [review]
Thanks for the update. The only thing I'm not sure is that if the SMT and predicate link is correct or not. Other than that, LGTM! Thank you.

**Comment by mimaison** [comment]
@dajac can you take a look? Thanks

File: docs/connect.html [hunk 2]
```diff
@@ -414,53 +403,49 @@ <h5><a id="connect_connectorexample" href="#connect_connectorexample">Connector
...
-        // Only one input stream makes sense.
-        Map&lt;String, String&gt; config = new HashMap&lt;&gt;();
-        if (filename != null)
-            config.put(FILE_CONFIG, filename);
-        config.put(TOPIC_CONFIG, topic);
-        configs.add(config);
-        return configs;
-    }
-    </pre>
+@Override
```
Comment (tombentley) [inline docs/connect.html:438]: Does putting this in a CDATA let us avoid the eye-bleeding HTML escaping?

File: connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java [hunk 3]
```diff
@@ -61,7 +61,7 @@ private static void printPredicateHtml(PrintStream out, DocInfo docInfo) {
         out.println("<div id=\"" + docInfo.predicateName + "\">");
 
         out.print("<h5>");
-        out.print(docInfo.predicateName);
+        out.print("<a href=\"#" + docInfo.predicateName + "\">" + docInfo.predicateName + "</a>");
```
Comment (tombentley) [inline connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java:64]: Are this really `href` not `name`? If so what are they linking to?

File: docs/connect.html (hunk 2 above)
Comment (mimaison) [reply docs/connect.html:438]: I'm not familiar with CDATA but from a quick search it seems it shouldn't be used in HTML documents.
https://developer.mozilla.org/en-US/docs/Web/API/CDATASection states:
```
Note: CDATA sections should not be used within HTML they are considered as comments and not displayed.
```

File: connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java (hunk 3 above)
Comment (mimaison) [reply connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java:64]: It's linking to the existing `<div id="...">` just above on line 61

**Comment by mimaison** [comment]
@showuon @tombentley Can you take another look? Thanks

File: connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java (hunk 3 above)
Comment (showuon) [reply connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java:64]: Could you let me know where does these `predicateName` appear in the doc?
here? https://kafka.apache.org/documentation/#connect_predicates
And what will be the expected output? What hyperlink destination we expected to link to?

Thank you.

**Review by showuon — COMMENTED** [review]
@mimaison , for the `html` changes, I think they are good to be merged. But for the doc tool part, I am still not sure if the change is good or not since I don't know how these tool work. Might need your help to explain to me.

Or, maybe we can first merge the `html` changes, and discuss the tools change in a separate PR. Up to you. Thanks.

File: connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java (hunk 3 above)
Comment (mimaison) [reply connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java:64]: Predicates are listed in this section: https://kafka.apache.org/documentation/#connect_predicates

Currently the names of predicates are listed but they are not links so we can't easily copy direct links to them.

Comment (showuon) [reply connect/runtime/src/main/java/org/apache/kafka/connect/tools/PredicateDoc.java:64]: OK, I got it. You want to add an anchor point in each predicate and transformation. I see now. I've checked, there is already the HTML element `id` = predicate names and transformation names, so adding an anchor link with the name can link to the position. Looks good to me. Thanks for the explanation.

**Review by showuon — APPROVED** [review]
LGTM! Thanks for the patch!

**Comment by showuon** [comment]
@tombentley , sorry, I found I forgot to ask you if you have other comments. If you have any comments, please raise it. Sorry!

**Pull request merged**